localstack/localstack
# implement lazy-loading of services

This PR adds

* A ServiceManager class that now takes care of managing AWS services (instead of previously having a only global arrays of state)
* A ServicePlugin concept that uses our plugin framework to load services
* optional lazy-loading of services 

To test:
* start localstack with `EAGER_SERVICE_START=0`
* run `localstack status services` and observe that services are in `available` mode

TODO: this will break localstack-ext. there's a branch with the same name in localstack-ext that needs to be updated and released at the same time.
